Enyim.Caching/Memcached/Locators/DefaultNodeLocator.cs

```diff
@@ -117,70 +104,48 @@ private IMemcachedNode Locate(string key)
             return Locate(key);
         }
 
+        public static ulong XorShiftMult64(ulong x)
+        {
+            // Combine operations for readability (C# allows for more relaxed expression formatting)
+            return x ^ (x >> 12) ^ (x << 25) ^ (x >> 27) * 2685821657736338717UL;
+        }
+
         /// <summary>
         /// locates a node by its key
         /// </summary>
         /// <param name="key"></param>
         /// <returns></returns>
         private IMemcachedNode FindNode(string key)
         {
-            if (_keys.Length == 0) return null;
+            if (_hrw.Nodes.Count == 0) return null;
 
-            uint itemKeyHash = BitConverter.ToUInt32(new FNV1a().ComputeHash(Encoding.UTF8.GetBytes(key)), 0);
-            // get the index of the server assigned to this hash
-            int foundIndex = Array.BinarySearch<uint>(_keys, itemKeyHash);
+            if (_hrw.Nodes.Count == 1) return _hrw.Nodes.First().Value;
 
-            // no exact match
-            if (foundIndex < 0)
-            {
-                // this is the nearest server in the list
-                foundIndex = ~foundIndex;
+            ulong keyHash = _hrw.Hash(key);
 
-                if (foundIndex == 0)
-                {
-                    // it's smaller than everything, so use the last server (with the highest key)
-                    foundIndex = _keys.Length - 1;
-                }
-                else if (foundIndex >= _keys.Length)
+            int nodeId = 0;
+
+            ulong maxHash = XorShiftMult64(keyHash ^ _hrw.Nhash[0]);
+
+            for (int i = 1; i < _hrw.Nhash.Count; i++)
+            {
+                ulong hash = XorShiftMult64(keyHash ^ _hrw.Nhash[i]);
+                if (hash > maxHash)
                 {
-                    // the key was larger than all server keys, so return the first server
-                    foundIndex = 0;
+                    nodeId = i;
+                    maxHash = hash;
```

Comment:
You have added this check to break the loop. We can simply right a break statement here. It will be more readable.